internal/ingress/controller/controller.go

```diff
@@ -185,18 +186,32 @@ func (n *NGINXController) syncIngress(interface{}) error {
 
 	isFirstSync := n.runningConfig.Equal(&ingress.Configuration{})
 	go func(isFirstSync bool) {
+		steps := 1
 		if isFirstSync {
-			glog.Infof("Initial synchronization of the NGINX configuration.")
-
-			// it takes time for NGINX to start listening on the configured ports
+			// For the initial sync it always takes some time for NGINX to
+			//  start listening on the configured port (default 18080)
+			//  For large configurations it might take a while so we loop
+			//  and back off
+			steps = 10
 			time.Sleep(1 * time.Second)
```

Comment:
I was wishing we could get rid of this with this PR.

Reply:
We definitely could, but there will always be at least 1 failure without the initial sleep. To avoid _always_ failing on the first attempt, I think we should leave in the initial 1 second sleep.

Reply:
@coreypobrien what do you think we get rid of `isFirstSync` completely? We can unconditionally set steps to 10.
IMO it's OK to have `configureDynamically` failing for few times during first sync.

That way we can simplify this code and also make sure we have retry logic for subsequent reconfigurations as well.